specifyweb/frontend/js_src/lib/components/Merging/index.tsx

```diff
@@ -246,17 +243,17 @@ function Merging({
                 errorMode: 'dismissible',
               }
             ).then(({ data, response }) => {
+              if (!response.ok) return;
               setMergeId(data);
-              if (!response.ok) {
-                setError(data);
-              } else {
-                return;
-              }
+              /*
+               * Because we can not pass down anything from the Query Builder
+               * as a prop, this is needed to rerun the query results once
+               * the merge completes.
+               * (the RecordMergingLink component is listening to the event)
```

Comment:
(not a request for change, just commenting)

> (the RecordMergingLink component is listening to the event)
...and in the future ORM version, more things could be listening for deleted event. That would allow to solve cases when i.e you delete a query, but query list dialog still shows it - though in the best case, the query dialog will just call `useCollection(...)` hook, and that hook will take care of fetcing, fetching more (for pagination), errors and listening for records being deleted